test.py

```diff
@@ -0,0 +1,180 @@
+# pyGene Fusion local testing
+import json
+import re
+from pydantic import BaseModel, validator, ValidationError, StrictStr, StrictInt, StrictBool
+from typing import Optional, List, Union
+
+class GeneDescriptor(BaseModel):
+    """Define a Gene, denoted by GeneDescriptor"""
+    type = 'GeneDescriptor'
+    value_id: str
+    label: StrictStr
+
+    @validator('value_id')
+    def is_valid(cls,v):
+        assert v.count(':') == 1 and v.find(' ') == -1, 'id must be a CURIE'
+        return v
+
+class SequenceLocation(BaseModel):
+    """Define SequenceLocation class"""
+    type = 'SequenceLocation'
+    seq_id: str
+    start: StrictInt
+    end: StrictInt
+
+    @validator('seq_id')
+    def is_curie(cls, v):
+        assert v.count(':') == 1 and v.find(' ') == -1, 'chr must be a CURIE'
+        return v
+
+class ChromosomeLocation(BaseModel):
+    """Define ChromosomeLocation class"""
+    type = 'ChromosomeLocation'
+    species: str
+    chr: str
+    start: str
+    end: str
+
+    @validator('species')
+    def is_curie(cls, v):
+        assert v.count(':') == 1 and v.find(' ') == -1, 'chr must be a CURIE'
+        return v
+
+    @validator('chr')
+    def valid_chr(cls, v):
+        assert v.isalnum(), 'chr must have only alphanumeric characters'
+        return v
```

Comment:
Should we allow for `X`, `Y` to be a valid chr?

Reply:
Yes

Reply:
Oh disregard this. I misread it as "alphabet".

Reply:
Oh disregard this. I misread it as "alphabet".